core/src/main/java/com/whylogs/core/ColumnProfile.java

```diff
@@ -31,28 +34,52 @@
 public class ColumnProfile {
   public static final int FREQUENT_MAX_LG_K = 7;
   private static final int CARDINALITY_LG_K = 12;
+  private static Map<String, String> ENV;
+  private static ImmutableList<String> nullEnvStrs;
 
   @NonNull private final String columnName;
   @NonNull private final CountersTracker counters;
   @NonNull private final SchemaTracker schemaTracker;
   @NonNull private final NumberTracker numberTracker;
   @NonNull private final ItemsSketch<String> frequentItems;
   @NonNull private final HllSketch cardinalityTracker;
+  private final ImmutableList<String> nullStrs;
+
+  static synchronized ImmutableList<String> nullStrsFromEnv() {
+    if (ColumnProfile.ENV == null) {
```

Comment:
no reason to store System.getenv in this environment variable

Reply:
ok, but I did is this way for a reason.  I wanted the normal case without any overrides to have a null value.
`this.nullStrs == null` to indicate there are no overrides.
Sounds like you prefer to always have a set of overrides, but most of the time the set is empty.

It was my assumption that testing `this.nullStrs == null`in the critical path is a helluva lot faster than calling `this.nullStrs.contains(v.toString())` even on an empty set.

Reply:
therefore I used `ColumnProfile.ENV` as a signal value of whether we had initialized the `nullStrs` set or not because null was a perfectly acceptable value for `nullStrs` so I could not use that as a signal.

Reply:
Performance wise it won't make a difference.

> therefore I used ColumnProfile.ENV as a signal value of whether we had initialized the nullStrs set or not because null was a perfectly acceptable value for nullStrs so I could not use that as a signal.

That's why you can just have `null` = uninitialized, and a non-null = `initialized`. And if you check on the volatile variable itself you won't need `synchronized`